core/utils/preflight.py

```
@@ -0,0 +1,293 @@
+            if available_gb < required_gb:
+                logger.error(f"❌ Insufficient disk space: {available_gb:.2f} GB available, {required_gb:.2f} GB required")
+                return False
+            
+            logger.info(f"✓ Disk space: {available_gb:.2f} GB available")
+            return True
+            
+        except Exception as e:
+            logger.error(f"❌ Could not check disk space: {e}")
+            return False
+    
+    def check_database_connection(self, host: str, database: str, 
```

Ruff (0.12.2)

Trailing whitespace

Remove trailing whitespace

(W291)